src/datachain/lib/file.py

```diff
@@ -432,14 +354,32 @@ class ImageFile(File):
 
     def read(self):
         """Returns `PIL.Image.Image` object."""
-        fobj = super().read()
-        return Image.open(BytesIO(fobj))
+        with self.open() as stream:
```

Comment:
hmm, am I missing something - or should it be the same (`super().read() still calls `self.open()`?)

Reply:
`super().read()` calls `super().open()`. So `LaionImageFile.read()` was calling `File.open()` instead of `TarVFile.open()` without this change.

Reply:
hmm, is it the same as:

```python
class Parent:
    def method1(self):
        print("Parent method1")

    def method2(self):
        print("Parent method2")
        self.method1()


class Child1(Parent):
    def method2(self):
        print("Child1 method2")
        super().method2()

class Child2(Parent):
    def method1(self):
        print("Child2 method1")


class Impl(Child1, Child2):
    pass

c = Impl()
c.method2()
```

it returns:

Child1 method2
Parent method2
Child2 method1

(so, even though we use super, it's important what is `self` in this case)?

Reply:
Sorry, guess I got confused here. I've reverted this change now.